source/slang/slang-ir-clone.cpp

```diff
@@ -181,10 +193,14 @@ static void _cloneInstDecorationsAndChildren(
         //
         if( oldChild->getFirstDecorationOrChild() )
         {
-            IRCloningOldNewPair pair;
-            pair.oldInst = oldChild;
-            pair.newInst = newChild;
-            pairs.add(pair);
+            pairs.add({ oldChild, newChild });
+        }
+
+        // Add params to a separate list so that we can clone their types later
+        // after the entire block is cloned.
+        if (oldChild->getOp() == kIROp_Param)
+        {
+            paramPairs.add({oldChild, newChild});
```

Comment:
Is there any reason that you can't put this at line 183?

Reply:
I can do that.